client.go

```diff
@@ -82,22 +83,34 @@ func NewDgraphClient(clients ...api.DgraphClient) *Dgraph {
 //		defer conn.Close()
 //		dgraphClient := dgo.NewDgraphClient(api.NewDgraphClient(conn))
 func DialCloud(endpoint, key string) (*grpc.ClientConn, error) {
-	u, err := url.Parse(endpoint)
-	if err != nil {
-		return nil, err
+	var grpcHost string
+	switch {
```

Comment:
Can you do this using a function instead? I think this would be useful in other places too. Also, can you provide a description of the different use cases that its checking for?

Reply:
I think we can do that when we need it. Let me add more comments to this function, that could be useful.